src/control/lib/hardware/libfabric/bindings.go

```diff
@@ -0,0 +1,124 @@
+//
+// (C) Copyright 2021 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package libfabric
+
+/*
+#cgo LDFLAGS: -lfabric
+#include <stdlib.h>
+#include <stdio.h>
+#include <rdma/fabric.h>
+#include <rdma/fi_domain.h>
+#include <rdma/fi_endpoint.h>
+#include <rdma/fi_cm.h>
+#include <rdma/fi_tagged.h>
+#include <rdma/fi_rma.h>
+#include <rdma/fi_errno.h>
+
+#define getHFIUnitError -2
+typedef struct {
+	uint64_t reserved_1;
+	uint8_t  reserved_2;
+	int8_t   unit;
+	uint8_t  port;
+	uint8_t  reserved_3;
+	uint32_t service;
+} psmx2_ep_name;
+
+int get_hfi_unit(void *src_addr) {
+	psmx2_ep_name *psmx2;
+	psmx2 = (psmx2_ep_name *)src_addr;
+	if (!psmx2)
+		return getHFIUnitError;
+	return psmx2->unit;
+}
+*/
+import "C"
+
+import (
+	"github.com/pkg/errors"
+)
+
+const (
+	libFabricMajorVersion = 1
+	libFabricMinorVersion = 7
+)
+
+func libFabricVersion() C.uint {
+	return C.uint(libFabricMajorVersion<<16 | libFabricMinorVersion)
+}
```

Comment:
Temporary code?

Reply:
We use it below, in `fiGetInfo()`. We have to pass the version into `fi_getinfo` when fetching all the fabric interfaces.

Reply:
I meant the hard-coded versions... Shouldn't those come from the library itself?

Reply:
Ah... I didn't dig too deeply into that. I harvested this from the old netdetect code directly.

Reply:
It looks like we can grab it from the library, you're right. Fixing.

Reply:
From the `fi_getinfo` manpage:

> The version parameter is used by the application to request the desired version of the interfaces. The version determines the format of all data structures used by any of the fabric interfaces. Applications should use the FI_VERSION(major, minor) macro to indicate the version, **with hard-coded integer values for the major and minor values**. The FI_MAJOR_VERSION and FI_MINOR_VERSION enum values defined in fabric.h specify the latest version of the installed library. However, it is recommended that the integer values for FI_MAJOR_VERSION and FI_MINOR_VERSION be used, rather than referencing the enum types in order to ensure compatibility with future versions of the library. This protects against the application being built from source against a newer version of the library that introduces new fields to data structures, which would not be initialized by the application.

Emphasis mine. It looks like hard-coding a specific version is recommended.

Reply:
I switched to using the macro to format it, but keeping the hard-coded versions for now, at least.

Reply:
Hmm, OK. So maybe we could rename them and/or put comments to make their usage clearer? If I'm understanding correctly, these versions indicate the desired (minimum?) version of the returned data structures. Is 1.7 a reasonable value, given that I think we're up to 1.14-ish now? Just wondering if we're missing some stuff from newer versions that would make our lives easier.

I'm fine with leaving things at the current version for the moment while we figure out if there's a better way to do this, but would prefer that the consts at least be named better.

Reply:
I killed the consts and shifted up into the Cgo block, with a comment. Is that adequate or should I add more detail?

Reply:
That's fine.